cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -128,6 +128,7 @@ predicate readStep(Node node1, Content f, Node node2) {
  * numeric conversions, and otherwise the erasure is used.
  */
 RefType getErasedRepr(Type t) {
+  t = t and // silence compiler warning
```

Comment:
There's a pattern of using `suppress*` predicates like `private predicate suppressUnusedThis(Function f) { any() }` for this, FWIW.

Reply:
I've added a commit to follow that convention.